packages/component/src/Composer.js

```diff
@@ -96,9 +96,8 @@ function createCardActionLogic({ directLine, dispatch }) {
         case 'playAudio':
         case 'playVideo':
         case 'showImage':
-          // TODO: [P3] We should support ponyfill for window.open
-          //       This is as-of v3
-          window.open(value);
+          const factory = this.props.externalUrlHandlerFactory({ value });
```

Comment:
@compulim - Let me know what you think should be included in the factory's function signature.

Reply:
How about?

`windowOpenPonyfillFactory({ activity, attachment, cardAction }) => (url: string, windowName: string, windowFeatures: string) => void`

For the object returned from the ponyfill, we can assume we are not acting on it, until we need it. So I am leaving it `void` right now.